apps/meteor/ee/app/livechat-enterprise/server/api/tags.ts

```diff
@@ -67,35 +69,58 @@ API.v1.addRoute(
 	},
 );
 
-const livechatTagsEndpoints = API.v1.post(
-	'livechat/tags.delete',
-	{
-		response: {
-			200: POSTLivechatTagsRemoveSuccessResponse,
-			400: validateBadRequestErrorResponse,
-			401: validateUnauthorizedErrorResponse,
-			403: validateForbiddenErrorResponse,
+const livechatTagsEndpoints = API.v1
+	.post(
+		'livechat/tags.save',
+		{
+			response: {
+				200: POSTLivechatTagsSaveSuccessResponse,
+				400: validateBadRequestErrorResponse,
+				401: validateUnauthorizedErrorResponse,
+				403: validateForbiddenErrorResponse,
+			},
+			authRequired: true,
+			permissions: ['manage-livechat-tags'],
+			license: ['livechat-enterprise'],
+			body: isPOSTLivechatTagsSaveParams,
 		},
-		authRequired: true,
-		permissions: ['manage-livechat-tags'],
-		license: ['livechat-enterprise'],
-		body: isPOSTLivechatTagsRemoveParams,
-	},
-	async function action() {
-		const { id } = this.bodyParams;
-		try {
-			await LivechatEnterprise.removeTag(id);
+		async function action() {
+			const { _id, tagData, tagDepartments } = this.bodyParams;
 
-			return API.v1.success();
-		} catch (error: unknown) {
-			if (error instanceof Meteor.Error) {
-				return API.v1.failure(error.reason);
-			}
+			const result = await LivechatEnterprise.saveTag(_id, tagData, tagDepartments);
 
-			return API.v1.failure('error-removing-tag');
-		}
-	},
-);
+			return API.v1.success(result);
+		},
+	)
+	.post(
+		'livechat/tags.delete',
+		{
+			response: {
+				200: POSTLivechatTagsRemoveSuccessResponse,
+				400: validateBadRequestErrorResponse,
+				401: validateUnauthorizedErrorResponse,
+				403: validateForbiddenErrorResponse,
+			},
+			authRequired: true,
+			permissions: ['manage-livechat-tags'],
+			license: ['livechat-enterprise'],
+			body: isPOSTLivechatTagsRemoveParams,
+		},
+		async function action() {
```

Comment:
Do you need the trycatch? I see you're catching the meteor error into an api response, but doesn't the try/catch on the api handler do that already? 👀